torii/src/routing.rs

```diff
@@ -151,15 +152,19 @@ pub async fn handle_pending_transactions(
     queue: Arc<Queue>,
     sumeragi: SumeragiHandle,
     pagination: Pagination,
-) -> Result<Scale<Vec<SignedTransaction>>> {
+    transaction: SignedTransaction,
+) -> Result<Scale<Option<SignedTransaction>>> {
     let query_response = sumeragi.apply_wsv(|wsv| {
         queue
             .all_transactions(wsv)
             .map(Into::into)
             .paginate(pagination)
```

Comment:
Hm, I don't think it makes sense to keep pagination in this endpoint. AFAIU, you can use it to specify a range in which to search for the transaction, which isn't really useful for the only function we use it for.

I would also suggest to remove the corresponding `get_original_transaction_with_pagination` on the client side, leaving all the implementation in `get_original_transaction`

Reply:
I agree that pagination here is a bit strange